scripts/files/fs_s3.py

```diff
@@ -32,13 +33,14 @@ def write(destination: str, source: bytes, content_type: Optional[str] = None) -
     s3_path = parse_path(destination)
     key = s3_path.key
     s3 = resource("s3")
+    fileChecksum = checksum.multihash_as_hex(source)
 
     try:
         s3_object = s3.Object(s3_path.bucket, key)
         if content_type:
-            s3_object.put(Body=source, ContentType=content_type)
+            s3_object.put(Body=source, ContentType=content_type, Metadata={"filechecksum": fileChecksum})
```

Comment:
It might be useful to call this something like "multihash" to make it more self-documenting. "file" is probably redundant in the metadata name.